lte/gateway/python/scripts/user_trace_cli.py

```diff
@@ -0,0 +1,302 @@
+        )
+
+        parser.add_argument(
+            "-p",
+            "--path",
+            type=Path,
+            default=Path(__file__).absolute().parent / "mme_apn.log",
+            help="Path to the data directory",
+        )
+
+        subparsers = parser.add_subparsers(dest="subcmd")
+
+        # list imsi and number of ocurrences in logs
+        list_imsi = subparsers.add_parser(
+            'list_imsi', help="List imsi and number of ocurrences in logs",
+        )
+        list_imsi.set_defaults(func=self.get_list_imsi)
+
+        # list mme and enb user id pairs
+        list_ue_id = subparsers.add_parser(
+            'list_ue_id', help="List mme and enb ue id pairs for an imsi",
+        )
+        list_ue_id.add_argument(
+            "-i", "--imsi", required=True,
+            help="IMSI of the subscriber",
+        )
+        list_ue_id.set_defaults(func=self.get_list_ue_id)
+
+        # output session trace based on imsi,  mme id
+        session_trace = subparsers.add_parser(
+            'session_trace', help="Dump session trace for given user id",
+        )
+        session_trace.add_argument(
+            "-m", "--mme_id", required=True,
+            help="mme id of the user", type=int,
+        )
+        session_trace.add_argument(
+            "-i", "--imsi", required=True,
+            help="IMSI of the subscriber",
+        )
+        session_trace.set_defaults(func=self.get_session_trace)
+
+        return parser
+
+    def get_list_imsi(self, args):
+        """List imsi and number of ocurrences in logs"""
+
+        imsi_pattern = r"\b\d{15}\b"
+
+        imsi_list = defaultdict(int)
+        lines = self.lines
+        for line in lines:
+            match = re.findall(imsi_pattern, line)
+            if not match:
+                continue
+            imsi = match[0]
+            imsi_list[imsi] += 1
+
+        sorted_imsi_list = sorted(imsi_list.items(), key=lambda x: x[1], reverse=True)
+
+        print("\n{0:<15} {1:<15} ".format('IMSI', 'Occurrences'))
+        print('-' * 30)
+
+        for v in sorted_imsi_list:
+            imsi, number = v
+            print("{0:<15} {1:<15}".format(imsi, number))
+
+        print("\nNumber of unique IMSI found in logs: " + str(len(imsi_list)))
+
+    def get_list_ue_id(self, args):
+        """
+        Provide list of mme id and enodeb id pairs from given an IMSI.
+        Some sessions may not have the pair(missing enodeb id). Getting mme_id from Attach Req
+
+        Args:
+            args: Filtered logs for a given imsi and mme id
+
+        Returns:
+            ue_id_list: list of mme id and enodeb id pairs from given an IMSI
+        """
+
+        hex_pattern = "0X.*"
+        pattern_mme_enb_ue_id = r".*MME_UE_S1AP_ID = (\w+)\b eNB_UE_S1AP_ID = (\w+)\b"
+        ue_id_list = []
+        mme_id_list = []
+        lines = self.lines
+        imsi = self.args.imsi
+        for line in lines:
+            ue_id = {}
```

Comment:
```suggestion
            ue_id_dict= {}
```
for clarity through the code